### PR TITLE
fix: add new tutorial for running VM based workloads

### LINKED ISSUE #254: feat: add new guide to run a VM workload using A4VM
### Description
- We need add a tutorial that shows running a VM based workload in Anthos bare metal using Anthos VMRuntime (i.e.: Kubevirt)

### PR DESCRIPTION
**Pull request by Shabirmean**
#### Description
- There is no tutorial/guide that shows deploying an existing VM based workload into an Anthos bare metal cluster using Anthos VMRuntime (i.e. Kubevirt)
- Thus, I have added a new tutorial that takes one through taking an existing VM Image (an image containing workloads) and spinning this VM inside a ABM cluster

#### Change summary
- Created a GCE VM
- Installed Point of Sale (PoS) sample application inside the GCE VM
- Created `systemd` services for spawning the PoS application upon GCE VM boot up
- Created a `sudo` user with username/password: `abmuser/abmpassword`
- Installed the `mysqlsh` CLI in the GCE VM in case we want to test connection
- Stopped the VM and [created an image of this VM](https://cloud.google.com/compute/docs/images/create-delete-deprecate-private-images#gcloud_1)
- [Exported this image](https://cloud.google.com/compute/docs/images/export-image#gcloud) to Google Cloud Storage in `qcow2` format and made it publicly accessible so people can pull the image
   - See image [here](https://storage.googleapis.com/abm-vm-images)
- Added the following resource files into this repo for visibility:
  - The `systemd` service files that were created inside  the VM
  - The `mysql-db` yaml file to create a MySQL instance
  - The `Service + Endpoint` yaml file to direct traffic from the Ingress to the VM
  - A sample of the autogenerated `VIrtualMachine` yaml file
- Finally, added step by step guide to put all of this together and get the PoS app running inside a A4VM managed VM in ABM. Whenever, someone creates a VM off of the above image in ABM, the PoS app will boot up by default.

#### Note to reviewers:
- The guideline for adding region tags is as follows:
   -  `<product_name>_<closest_enclosing_folder/directory>_<file_name>_<krm_resource>_<krm_resource_name>`
   - All hyphens are turned into underscores
- Please use the above guideline to review the newly added region tags.
- See [rendered guide here](https://github.com/GoogleCloudPlatform/anthos-samples/tree/kubevirt-guide/anthos-vmruntime)

#### Related PRs/Issues
- N/A

### CODE REVIEW

**Comment by ckim328** [comment]
praise: great job region tagging!! So thorough!! 
Once the tests pass, i'll stamp 

small question: I see in this file anthos-vmruntime/pos-systemd-services/pos_apiserver.service that you have your name there as User, I'm guessing it's a basic mock up?

**Comment by Shabirmean** [comment]
> praise: great job region tagging!! So thorough!! I don't have time to run though the whole setup till later this week, but i give my approval for the region tags 👏
> 
> small question: I see in this file anthos-vmruntime/pos-systemd-services/pos_apiserver.service that you have your name there as User, I'm guessing it's a basic mock up?

Thank you very much for the swift review. Yes sure, take your time to come back to it whenever you have time. Good catch about the `User`. It shouldn't be using my name. I have updated it in: https://github.com/GoogleCloudPlatform/anthos-samples/pull/255/commits/4a7cf66395adccd2c0191281288ee6a362bd9b0d

File: anthos-vmruntime/README.md [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+by setting the `spec.enabled` field to `true`.
+```sh
+kubectl edit vmruntime
+```
+ You can also copy the [vmruntime.yaml](vmruntime.yaml)
+file from this repository into your admin workstation and directly apply that
+using `kubectl apply -f vmruntime.yaml`.
+
+> **Note:** Optionally, you can set the `spec.useEmulation` field to `true`.
+> Setting this to true ensures that the VMRuntime makes use of hardware
```
Comment (apeabody) [inline anthos-vmruntime/README.md:97]: @Shabirmean - I believe that setting to `true` actually permits the use of *software* emulation.  While hardware emulation should always be attempted first, if this is `false` then *software* emulation is NOT permitted.

https://github.com/GoogleCloudPlatform/anthos-samples/blame/kubevirt-guide/anthos-vmruntime/README.md#L96

File: anthos-vmruntime/README.md [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+bare metal clusters since version `1.10`. An instance of the `VMRuntime` custom
+resource is already created upon installation. However, it is `disabled` by
+default. Thus, you have to `enable` it to be able to create and manage VMs using
+the `Anthos VMRuntime`.
+
+The following command would open the resource specification for the `VMRuntime`
+instance in the default editor of your shell session. Update the specification
+by setting the `spec.enabled` field to `true`.
+```sh
+kubectl edit vmruntime
```
Comment (apeabody) [inline anthos-vmruntime/README.md:90]: OPTIONAL - You might like a step after to confirm that the VM Runtime is enabled:

`kubectl describe vmruntime vmruntime | grep Enabled`

Enabled:                       true

File: anthos-vmruntime/README.md [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+
+When you delete the VM, the associated resources are also deleted except for the
+`DataVolume` that gets created to import/host the VM image. You can manually
+delete this resource. You can also use it to mount as a boot volume into new VM.
+
+```sh
+kubectl virt delete vm pos-vm
+
+# if you want to also delete the datavolume
+kubectl delete dv pos-vm-dv
```
Comment (apeabody) [inline anthos-vmruntime/README.md:404]: @Shabirmean I think this should be `kubectl delete dv pos-vm-boot-dv`?

Comment (Shabirmean) [reply anthos-vmruntime/README.md]: Fixed in https://github.com/GoogleCloudPlatform/anthos-samples/pull/255/commits/1b21d191f2b3ddeeb42f2853b6d895307af9010e

File: anthos-vmruntime/README.md (hunk 2 above)
Comment (Shabirmean) [reply anthos-vmruntime/README.md:90]: Added in https://github.com/GoogleCloudPlatform/anthos-samples/pull/255/commits/1b21d191f2b3ddeeb42f2853b6d895307af9010e

File: anthos-vmruntime/README.md (hunk 1 above)
Comment (Shabirmean) [reply anthos-vmruntime/README.md]: Updated in https://github.com/GoogleCloudPlatform/anthos-samples/pull/255/commits/1b21d191f2b3ddeeb42f2853b6d895307af9010e

File: anthos-vmruntime/README.md (hunk 2 above)
Comment (Shabirmean) [reply anthos-vmruntime/README.md:90]: Changed in 189a8798b2237f0ae4e46c5a37caacdd6681a515

**Comment by Shabirmean** [comment]
@apeabody Can you TAL at the changes?

**Review by apeabody — COMMENTED** [review]
LGTM!

**Comment by Shabirmean** [comment]
> LGTM!

@apeabody  Thank you. Can you click the `Add your review`-> `Review changes` -> `Approve` so that I am to merge

**Review by apeabody — APPROVED** [review]
👍

**Review by ckim328 — APPROVED** [review]
LGTM!

**Pull request merged**